### PR TITLE
Update to Linter 3 + fixes the messages not being updated

### PR DESCRIPTION
**Pull request by aminya**
This updates minimap-linter so it is in sync with linter 3 changes. It fixes the messages not being updated after changes.

The issue of minimap-linter was because JavaScript cannot compare complex objects well, and so `Set.add(message)` or `Set.delete(message)`  do not add/delete messages! I changed the code to use `Map` with my custom `key` (`message.key`).

### CODE REVIEW

**Comment by aminya** [comment]
@Arcanemagus Could you review this?

File: package.json [hunk 1]
```diff
@@ -79,7 +79,7 @@
     }
   },
   "package-deps": [
-    "linter:2.1.1"
+    "linter:3.0.0"
```
Comment (Arcanemagus) [inline package.json:82]:
```suggestion
    "linter:2.1.1"
```
`linter@3` isn't required at all, `message.key` was added to `linter` _waaay_ back in v2 [here](https://github.com/steelbrain/linter/commit/aa4c48d84546bbb1abe47674364d1bb87f2ed3ae).

(Obviously don't just accept this change, simply remove the commit that added this.)

**Review by Arcanemagus — CHANGES_REQUESTED** [review]
The commits either need to be squashed into a single commit for these changes, or they need to be re-ordered so that each commit comes in a sensible order. As it stands right now you have many commits that are dependent on changes in commits that come _after_ them.

These commits are all currently tagged with `feat:` when they should be `fix:` since nothing has actually changed with `linter`, this is just fixing a bug here.

File: package.json (hunk 1 above)
Comment (aminya) [reply package.json]: Fixed in https://github.com/AtomLinter/atom-minimap-linter/pull/82/commits/c267b22e5be996e0cbf3c979ae55bb51d4ad77fa

**Comment by aminya** [comment]
> The commits either need to be squashed into a single commit for these changes, or they need to be re-ordered so that each commit comes in a sensible order. As it stands right now you have many commits that are dependent on changes in commits that come _after_ them.
> 
> These commits are all currently tagged with `feat:` when they should be `fix:` since nothing has actually changed with `linter`, this is just fixing a bug here.

You can squash and merge the messages with your preferred message. 

image

**Pull request merged**